lms/djangoapps/save_for_later/api/v1/views.py

```diff
@@ -59,79 +54,76 @@ def post(self, request):
         data = request.data
         course_id = data.get('course_id')
         email = data.get('email')
-        marketing_url = data.get('marketing_url')
-        org_img_url = data.get('org_img_url')
+
+        if getattr(request, 'limited', False):
+            return Response({'error_code': 'rate-limited'}, status=403)
+
+        if get_email_validation_error(email):
+            return Response({'error_code': 'incorrect-email'}, status=400)
 
         course_key = CourseKey.from_string(course_id)
+        course = get_course(course_key, user.id, email)
```

Comment:
It will raise `CourseOverview.DoesNotExist` exception for the non-existent course and won't work as expected.